internal/cloud/aws/logger.go

```diff
@@ -6,20 +6,192 @@ SPDX-License-Identifier: AGPL-3.0-only
 
 package aws
 
-// TODO: Implement for AWS.
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+	logs "github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+	"k8s.io/utils/clock"
+)
 
 // Logger is a Cloud Logger for AWS.
-type Logger struct{}
+// Log messages are collected and periodically flushed to AWS Cloudwatch Logs.
+type Logger struct {
+	api logAPI
+
+	groupName  string
+	streamName string
+
+	logs          []types.InputLogEvent
+	sequenceToken *string
+
+	flushMux sync.Mutex
+	interval time.Duration
+	clock    clock.WithTicker
+	wg       *sync.WaitGroup
+	stopCh   chan struct{}
+}
 
 // NewLogger creates a new Cloud Logger for AWS.
-func NewLogger() *Logger {
-	return &Logger{}
+func NewLogger() (*Logger, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithEC2IMDSRegion())
+	if err != nil {
+		return nil, err
+	}
+	client := logs.NewFromConfig(cfg)
+
+	l := &Logger{
+		api:      client,
+		interval: time.Second,
+		clock:    clock.RealClock{},
+		wg:       &sync.WaitGroup{},
+		stopCh:   make(chan struct{}, 1),
+	}
+
+	if err := l.createStream(imds.New(imds.Options{})); err != nil {
+		return nil, err
+	}
+
+	l.wg.Add(1)
+	go l.flushLoop()
```

Comment:
```suggestion
	l.startFlushLoop()
```
Imo it is better to keep the concurrency stuff in one place.